extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSources.java

```diff
@@ -274,8 +280,9 @@ private void applyNewConfiguration(AgroalDataSourceConfigurationSupplier dataSou
             TransactionIntegration txIntegration = new NarayanaTransactionIntegration(transactionManager,
                     transactionSynchronizationRegistry, null, false,
                     dataSourceJdbcBuildTimeConfig.transactions == io.quarkus.agroal.runtime.TransactionIntegration.XA
-                            ? xaResourceRecoveryRegistry
-                            : null);
+                            && transactionRuntimeConfig.enableXARecovery
```

Comment:
@gsmet should we log a WARNn message here if `quarkus.datasource.jdbc.transaction=XA` but recovery is not enable  on startup?

Reply:
What you could do is to keep the boolean but make it an Optional one and enable recovery by default if XA is enabled.

Thus, if XA is enabled, you have the default behavior and if someone really wants to disable it, they can. By doing this, you wouldn't need a warning.

Reply:
Ah, except I see enabling XA is part of Agroal and not the Narayana extension.

Wait for a few minutes, I'm going to post a proper actionable review so that we get out of this situation for 2.13 without reverting your patch.